### PR TITLE
Flicker in default dark theme

### LINKED ISSUE #1451: Flickering in default dark theme
### Description

There's flickering in default dark theme on reload or while navigating to other pages of site.

https://github.com/meshery/meshery.io/assets/110674407/c3cc5eaf-642d-443d-90b7-99e71431254d

### Expected Behavior

 The light theme doesn't have any flickering issue and the same is desired for the dark-theme too.

### Enviroment:
 - OS: Mac Linux Windows
 - Browser: Chrome Safari Firefox
 - Version: 
 - Device: Desktop Mobile

---
## Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

The meshery.io website uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/meshery/meshery.io/tree/master).
- 📚 See the [Contributing to Meshery.io Website](https://github.com/meshery/meshery.io#contributing-to-the-mesheryio-website) section of the readme.md and other [contributing instructions](https://docs.meshery.io/project/contributing), too.
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by coder12git**
**Description**

This PR fixes #1451

**Notes for Reviewers**

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by abhijeetgauravm** [comment]
Let's discuss this on website call today at 5:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima)

**Comment by coder12git** [comment]
cc @sudhanshutech I followed all steps similar to what was sone to fix the flickering issue for docs, but still it's not fixed. Let me know what I am doing wrong!

**Comment by iArchitSharma** [comment]
@coder12git have you figured out if this is happening due to CSS Rendering or JavaScript Timing?

**Comment by coder12git** [comment]
> @coder12git have you figured out if this is happening due to CSS Rendering or JavaScript Timing?

Yep mainly due to js timing, that's why I added the main logic in starting of the body tag.

**Comment by leecalcote** [comment]
Please remove the draft label when ready for another review.

**Comment by iArchitSharma** [comment]
@coder12git I have not seen what JS logic has been used but I got some idea by seeing the `_includes/header.html` 
The Initial State of Dark mode is not set correctly, the initial state is set to the light mode which i guess is being manipulated by JS
So I guess we have to set the correct initial state in HTML and optimizing JavaScript logic

**Comment by coder12git** [comment]
> @coder12git I have not seen what JS logic has been used but I got some idea by seeing the `_includes/header.html` The Initial State of Dark mode is not set correctly, the initial state is set to the light mode which i guess is being manipulated by JS So I guess we have to set the correct initial state in HTML and optimizing JavaScript logic

Hmm i agree, can be fix by using dark-mode class in body which I tried but it doesn't work

**Comment by iArchitSharma** [comment]
> > @coder12git I have not seen what JS logic has been used but I got some idea by seeing the `_includes/header.html` The Initial State of Dark mode is not set correctly, the initial state is set to the light mode which i guess is being manipulated by JS So I guess we have to set the correct initial state in HTML and optimizing JavaScript logic
> 
> Hmm i agree, can be fix by using dark-mode class in body which I tried but it doesn't work, will see again :)

I know that because you not only have to change the 'dark-mode' class in the body but also all the 'images/logos' to their dark versions. I tried it, and it worked; there is no more flickering. However, there is still a problem: the icon of theme toggler itself is not changing on switching it.

**Comment by coder12git** [comment]
Hey @iArchitSharma let's discuss this in slack thread if you want, as this bug need to be fix as soon as possible.

**Comment by abhijeetgauravm** [comment]
Let's discuss this on website call tomorrow at 5:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima).

File: _layouts/home.html
Comment (vishalvivekm) [inline _layouts/home.html:1]: why are we changing home layout ?

**Comment by vishalvivekm** [comment]
@coder12git @iArchitSharma 
Good work, here.

Good that you've tried to split the code into components and then accordingly putting them in the over-all flow.
I, however, wonder just this one change ( the logic for checking the current set theme and applying effects acc to it)  below in `_includes/header.html` should fix the issue: 
### Current
https://github.com/meshery/meshery.io/blob/6afdd371515c66fede87dbb1e380ef35c4ec039c/_includes/header.html#L55

### After the change:
```
  <body style="overflow-x: hidden" class="dark-mode">
    <script>
      const body = document.body;
  
      if (localStorage.getItem("mode") === "light-mode") {
        body.classList.remove("dark-mode");
  
        let allLogos = document.querySelectorAll("#logo-dark-light");
        allLogos.forEach(e => e.src = e.dataset.logoForLight);
      }
  
      // Automatically set logos based on initial theme
      window.addEventListener('load', () => {
        let allLogos = document.querySelectorAll("#logo-dark-light");
        allLogos.forEach(e => {
          e.src = body.classList.contains("dark-mode") ? e.dataset.logoForDark : e.dataset.logoForLight;
        });
      });
    </script>

**Comment by coder12git** [comment]
> @coder12git @iArchitSharma 
> Good work, here.
> 
> Good that you've tried to split the code into components and then accordingly putting them in the over-all flow.
> I, however, wonder just this one change ( the logic for checking the current set theme and applying effects acc to it)  below in `_includes/header.html` should fix the issue: 
> ### Current
> https://github.com/meshery/meshery.io/blob/6afdd371515c66fede87dbb1e380ef35c4ec039c/_includes/header.html#L55
> 
> ### After the change:
> ```
>   <body style="overflow-x: hidden" class="dark-mode">
>     <script>
>       const body = document.body;
>   
>       if (localStorage.getItem("mode") === "light-mode") {
>         body.classList.remove("dark-mode");
>   
>         let allLogos = document.querySelectorAll("#logo-dark-light");
>         allLogos.forEach(e => e.src = e.dataset.logoForLight);
>       }
>   
>       // Automatically set logos based on initial theme
>       window.addEventListener('load', () => {
>         let allLogos = document.querySelectorAll("#logo-dark-light");
>         allLogos.forEach(e => {
>           e.src = body.classList.contains("dark-mode") ? e.dataset.logoForDark : e.dataset.logoForLight;
>         });
>       });
>     </script>
>     

No @vishalvivekm , we actually tried this, but flicker was still happening in both modes, just the duration was changed to few seconds, that's why me and archit cane to a conclusion that we need to restructure the entire codd to fix this :)

**Comment by vishalvivekm** [comment]
> @vishalvivekm , so should we revert our pr and only change that one file or the current one is fine?

I remember, on websites call y'day @iArchitSharma saying he'll be PRing the changes he has done locally

**Comment by iArchitSharma** [comment]
@vishalvivekm, done. I hope everything is fine now

**Review by vishalvivekm — APPROVED** [review]
Thanks @coder12git @iArchitSharma 🚀 
LGTM. // @sudhanshutech

**Review by vishalvivekm — APPROVED** [review]
Thanks @coder12git @iArchitSharma 🚀 
LGTM. // @sudhanshutech

**Pull request merged**